app/src/main/kotlin/org/fossify/clock/models/Timer.kt

```diff
@@ -17,7 +19,55 @@ data class Timer(
     var createdAt: Long,
     var channelId: String? = null,
     var oneShot: Boolean = false,
-)
+) : JSONConvertible {
+    override fun toJSON(): String {
+        val jsonObject = JSONObject()
+        jsonObject.put("id", id)
+        jsonObject.put("seconds", seconds)
+        jsonObject.put("vibrate", vibrate)
+        jsonObject.put("soundUri", soundUri)
+        jsonObject.put("soundTitle", soundTitle)
+        jsonObject.put("label", label)
+        jsonObject.put("createdAt", createdAt)
+        jsonObject.put("oneShot", oneShot)
+        return jsonObject.toString()
+    }
+
+    companion object {
+        fun parseFromJSON(jsonObject: JSONObject): Timer? {
+
+            if (!jsonObject.has("id") ||
+                !jsonObject.has("seconds") ||
+                !jsonObject.has("vibrate") ||
+                !jsonObject.has("soundUri") ||
+                !jsonObject.has("soundTitle") ||
+                !jsonObject.has("label") ||
+                !jsonObject.has("createdAt")
+            ) {
+                return null
+            }
+
+            val id = jsonObject.getInt("id")
+            val second = jsonObject.getInt("seconds")
+            val vibrate = jsonObject.getBoolean("vibrate")
+            val soundUri = jsonObject.getString("soundUri")
+            val soundTitle = jsonObject.getString("soundTitle")
+            val label = jsonObject.getString("label")
+            val createdAt = jsonObject.getLong("createdAt")
+
+            return Timer(
+                id,
+                second,
+                TimerState.Idle,
+                vibrate,
+                soundUri,
+                soundTitle,
+                label,
+                createdAt
+            )
+        }
+    }
+}
```

Comment:
We don't need to handle all this manually. Use `@kotlinx.serialization.Serializable` annotation, it'll be much simpler that way.

You can check how how it's done in Phone:

- https://github.com/FossifyOrg/Phone/blob/master/app/src/main/kotlin/org/fossify/phone/activities/SettingsActivity.kt
- https://github.com/FossifyOrg/Phone/blob/master/app/src/main/kotlin/org/fossify/phone/models/RecentCall.kt